packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -742,7 +743,9 @@ export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget
         super.dispose();
     }
 
-    protected resizeTerminal(): void {
+    protected resizeTerminal = debounce(() => this.doResizeTerminal(), 0);
```

Comment:
I've noticed that with a 0ms delay, the view still flickers on FireFox. [VSCode is using a delay of 50ms](https://github.com/microsoft/vscode/blob/e7155a90da4d62c3267e6fb96f56f176ed5f2a1b/src/vs/workbench/contrib/terminal/browser/terminalInstance.ts#L1863-L1866), we should use that as well.

Afterwards, this looks good to go 👍